pkg/engine/context/loaders/globalcontext.go

```diff
@@ -106,48 +106,20 @@ func (g *gctxLoader) loadGctxData() ([]byte, error) {
 		return nil, err
 	}
 
-	var jsonData []byte
-	if _, ok := data.([]byte); ok {
-		jsonData = data.([]byte)
-	} else {
-		jsonData, err = json.Marshal(data)
+	if rc.JMESPath == "" {
+		jsonData, err := json.Marshal(data)
 		if err != nil {
 			return nil, err
 		}
-	}
-	g.logger.V(6).Info("fetched json data", "name", g.entry.Name, "jsondata", jsonData)
-
-	if g.entry.GlobalReference.JMESPath == "" {
-		err := g.enginectx.AddContextEntry(g.entry.Name, jsonData)
-		if err != nil {
-			g.logger.Error(err, "failed to add resource data to context entry")
-			return nil, fmt.Errorf("failed to add resource data to context entry %s: %w", g.entry.Name, err)
-		}
-
 		return jsonData, nil
 	}
 
-	path, err := variables.SubstituteAll(g.logger, g.enginectx, rc.JMESPath)
```

Comment:
What if `rc.JMESPath` has variables?

Reply:
https://github.com/kyverno/kyverno/blob/v1.13.4/pkg/engine/context/loaders/globalcontext.go#L85-L89
https://github.com/kyverno/kyverno/blob/v1.13.4/pkg/engine/context/loaders/globalcontext.go#L124-L128

This is duplicate code too, keep [L85-L89](https://github.com/kyverno/kyverno/blob/v1.13.4/pkg/engine/context/loaders/globalcontext.go#L85-L89) and remove [L124-L128](https://github.com/kyverno/kyverno/blob/v1.13.4/pkg/engine/context/loaders/globalcontext.go#L124-L128) to support name use template variables.

Reply:
<img width="798" alt="image" src="https://github.com/user-attachments/assets/1650d8b6-2e77-4fe4-9d75-731476479f10" />

Reply:
Missed it, thanks!